File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.go
func isOneBitCharacter(bits []int) bool {
	n := len(bits)
	for i := 0; i < n; i++ {
		flag := false
		if bits[i] == 1 {
			i++
			flag = true
		}
		if i == n-1 && flag {
			return false
		}
	}

	return true
}

File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.cpp
class Solution {
   public:
    bool isOneBitCharacter(vector<int>& bits) {
        int n = bits.size();
        for (int i = 0; i < n; i++) {
            bool flag = false;
            if (bits[i] == 1) {
                i++;
                flag = true;
            }
            if (i == n - 1 && flag) {
                return false;
            }
        }

        return true;
    }
};